src/libstore/build/derivation-goal.cc

```diff
@@ -444,8 +444,6 @@ Goal::Co DerivationGoal::gaveUpOnSubstitution()
     trace("all inputs realised");
 
     if (nrFailed != 0) {
-        if (!useDerivation)
-            throw Error("some dependencies of '%s' are missing", worker.store.printStorePath(drvPath));
```

Comment:
I think the idea was that in the 0-dep case, this is more of an internal error. I think that's a small detail and it's fine to just use the other error code-path regardless?

See change originating in 1511aa9f488ba0762c2da0bf8ab61b5fde47305d